spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -145,13 +146,15 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		Resource metadataResource = appRegistration.getMetadataResource();
 
 		TaskExecution taskExecution = taskExecutionRepository.createTaskExecution();
-		taskDefinition = this.updateTaskProperties(taskDefinition, taskExecution);
+		taskDefinition = this.updateTaskProperties(taskDefinition);
 
 		Map<String, String> appDeploymentProperties = extractAppProperties(taskDefinition.getRegisteredAppName(), taskDeploymentProperties);
 		Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils.extractAndQualifyDeployerProperties(taskDeploymentProperties, taskDefinition.getRegisteredAppName());
 		AppDefinition revisedDefinition = mergeAndExpandAppProperties(taskDefinition, metadataResource, appDeploymentProperties);
 
-		AppDeploymentRequest request = new AppDeploymentRequest(revisedDefinition, appResource, deployerDeploymentProperties, commandLineArgs);
+		List<String> updatedCmdLineArgs = new ArrayList<>(commandLineArgs);
```

Comment:
Should we check to see if the spring.cloud.task.executionid is already present in the command line args.  If already present should we throw an exception, since Data Flow needs to own it (at least for CF).